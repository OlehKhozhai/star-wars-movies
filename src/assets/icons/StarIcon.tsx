import { FC } from "react";

type StarIconProps = {
  fill?: string;
  className?: string;
};

export const StarIcon: FC<StarIconProps> = ({ fill = "grey", className = "" }) => (
  <svg className={className} height="25" width="23" fill={fill}>
    <polygon
      points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
      style={{ fillRule: "nonzero" }}
    />
  </svg>
);
