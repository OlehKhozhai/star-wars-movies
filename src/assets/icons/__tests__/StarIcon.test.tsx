import { create } from "react-test-renderer";

import { StarIcon } from "../StarIcon";

describe("StarIcon component", () => {
  it("should render by default", () => {
    const component = create(<StarIcon />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should render with props properly", () => {
    const component = create(<StarIcon fill="green" className="test" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
