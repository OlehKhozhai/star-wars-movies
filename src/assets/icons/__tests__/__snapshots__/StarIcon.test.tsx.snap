// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarIcon component should render by default 1`] = `
<svg
  className=""
  fill="grey"
  height="25"
  width="23"
>
  <polygon
    points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
    style={
      Object {
        "fillRule": "nonzero",
      }
    }
  />
</svg>
`;

exports[`StarIcon component should render with props properly 1`] = `
<svg
  className="test"
  fill="green"
  height="25"
  width="23"
>
  <polygon
    points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
    style={
      Object {
        "fillRule": "nonzero",
      }
    }
  />
</svg>
`;
